src/Aspire.Hosting/ApplicationModel/ManifestMetadataAnnotation.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an entry in the metadata property of a generated manifest. 
+/// </summary>
+/// <param name="name"></param>
+/// <param name="value"></param>
+public class ManifestMetadataAnnotation(string name, object value) : IResourceAnnotation
+{
+    /// <summary>
+    /// Name of metadata entry.
+    /// </summary>
+    public string Name { get; } = name;
+
+    /// <summary>
+    /// Value of metadata entry.
+    /// </summary>
+    public object Value { get; } = value;
```

Comment:
string?

Reply:
On hold for now but based on latest offline chat we might still allow object serialization.